Alloy/Renderers/LinesRenderer.swift

```diff
@@ -0,0 +1,140 @@
+    public var lines: [Line] {
+        set {
+            var lines = newValue
+            self.linesCount = lines.count
+            self.linesBuffer = self.renderPipelineState.device
+                .makeBuffer(bytes: &lines,
+                            length: MemoryLayout<Line>.stride * lines.count,
+                            options: .storageModeShared)
+        }
+        get {
+            if let linesBuffer = self.linesBuffer,
+               let lines = linesBuffer
+                   .array(of: Line.self,
+                          count: self.linesCount) {
+                return lines
+            } else {
+                return []
+            }
+        }
+    }
+    /// Lines color. Red in default.
+    public var color: vector_float4 = .init(1, 0, 0, 0)
+
+    private var linesBuffer: MTLBuffer?
+    private var linesCount: Int = 0
+
+    private let renderPipelineState: MTLRenderPipelineState
+
+    // MARK: - Life Cycle
+
+    /// Creates a new instance of LinesRenderer.
+    ///
+    /// - Parameters:
+    ///   - context: Alloy's Metal context.
+    ///   - pixelFormat: Color attachment's pixel format.
+    /// - Throws: Library or function creation errors.
+    public convenience init(context: MTLContext, pixelFormat: MTLPixelFormat = .bgra8Unorm) throws {
+        guard
+            let library = context.shaderLibrary(for: LinesRenderer.self)
+        else { throw Errors.libraryCreationFailed }
+
+        try self.init(library: library, pixelFormat: pixelFormat)
+    }
+
+    /// Creates a new instance of LinesRenderer.
+    ///
+    /// - Parameters:
+    ///   - library: Alloy's shader library.
+    ///   - pixelFormat: Color attachment's pixel format.
+    /// - Throws: Function creation error.
+    public init(library: MTLLibrary, pixelFormat: MTLPixelFormat = .bgra8Unorm) throws {
+        guard
+            let vertexFunction = library.makeFunction(name: LinesRenderer.vertexFunctionName),
+            let fragmentFunction = library.makeFunction(name: LinesRenderer.fragmentFunctionName)
+        else { throw Errors.functionCreationFailed }
+
+        let renderPipelineDescriptor = MTLRenderPipelineDescriptor()
+        renderPipelineDescriptor.vertexFunction = vertexFunction
+        renderPipelineDescriptor.fragmentFunction = fragmentFunction
+        renderPipelineDescriptor.colorAttachments[0].pixelFormat = pixelFormat
+
+        self.renderPipelineState = try library.device
+            .makeRenderPipelineState(descriptor: renderPipelineDescriptor)
+    }
+
+    // MARK: - Rendering
+
+    /// Render lines in a target texture.
+    ///
+    /// - Parameters:
+    ///   - renderPassDescriptor: Render pass descriptor to be used.
+    ///   - commandBuffer: Command buffer to put the rendering work items into.
+    public func render(renderPassDescriptor: MTLRenderPassDescriptor,
+                       commandBuffer: MTLCommandBuffer) throws {
+        #if DEBUG
+        // Check render target.
```

Comment:
I don't think this check is needed, Metal does it internally with Metal API Validation